src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -599,6 +599,34 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     }
   }
 
+  /**
+   * Bind two threads together so that if one is blocked the other should be blocked too.
+   */
+  void bind_propagate_threads(long const first, long const second)
```

Comment:
See https://github.com/NVIDIA/spark-rapids/pull/12312#discussion_r1995602239

I would rather flip this around and have the default be any blocked, and inform the state machine of thread pool threads instead.